### ISSUE TITLE
Generate a second property for the URI path computationally

### ISSUE THREAD

**Issue by toddpadwick** [hdoro/sanity-plugin-prefixed-slug#10]
This is a great plugin (definitely should be core). however, I do have one improvement that would be really helpful for many of my projects:
With the functionality you've created to create the prefix from elements from the document, it would be great to automatically generate a second property named 'uri' or 'fullPath', with the prefix and slug combined. This way you can access both the slug and the full path of the document in the API. Eg you would get something like:
```
{
   _type:'podcast',
   slug:'some-podcast-title',
   fullPath:'blog/podcasts/some-category-title/some-podcast-title',
   blogCategory: {
        title:'Some Category title',
        slug:'some-category-title'
   }
}
```

**Comment by hdoro** [comment]
Thank you for your suggestion, Todd!

In the v3 version of the plugin (`^2.0.0`), I've added a `storeFullUrl` boolean flag for those looking to save the prefixed slug in the Sanity dataset. It'll show up in the slug object under the `fullUrl` property - here's an example:

```js
{ _type: "slug", current: "my-slug", fullUrl: "https://site.com/my-slug" }
```

This also fixes #4 , which is essentially the same request.

**Comment by toddpadwick** [comment]
This is fantastic, thanks @hdoro ! so much better!. However, still one slight problem that could be improved with the following:
in Sanity studio, you may want to have the domain in the prefix along with a little link button so the user can quickly access the page on the live site and see its a full URL. However, in the API, you'd need the full path but without the domain,  otherwise, it will cause issues with multi-environment setups (staging etc).  so something like this would be awesome:

config schema:
```
{
 name: 'slug_function_path',
 type: 'slug',
  options: {
    domain:process.env.SITE_URL,
    pathPrefix: (document) => `blog/${document.blogCategory}`,
   source:title
  },
```

CMS graphical interface view:
```
My Slug field
 ___________________________________________________________________
|                                       |               |          |
|   https://www.mysite.com/blog/news/   |    my-slug    |    ==>   |
|_______________________________________|_______________|__________|
(visible prefix)                         (slug input)    (Hyperlink to full url)

```

API output
```
{ 
_type: "slug",
current: "my-slug",
path: "/blog/news/my-slug", // path only without domain
url:"https://www.mysite.com/blog/news/my-slug" // probably not needed by no harm in having 
}
```
